tasks/extract_filepaths_for_granule/extract_filepaths_for_granule.py

```diff
@@ -78,20 +78,23 @@ def task(event, context):  # pylint: disable-msg=unused-argument
                     LOGGER.debug(f"File {file_name} will be restored")
                     file_key = afile["key"]
                     LOGGER.debug(f"Retrieving information for {file_key}")
-                    dest_bucket = None
-                    for key in regex_buckets:
```

Comment:
Since this loop processed every element, regardless of what was found, later matches could replace earlier ones. This does not feel intentional, and doesn't seem to be documented. If we wish to preserve it, I could.